### PR TITLE
feat: upgrade typescript version to 5.3.2

### PR DESCRIPTION
**Pull request by marcus-sa**
### Relinquishment of Rights

Please mark following checkbox to confirm that you relinquish all rights of your changes:

- [x] I waive and relinquish all rights regarding this changes (including code, text, and images) to Deepkit UG (limited), Germany. This changes (including code, text, and images) are under MIT license without name attribution, copyright notice, and permission notice requirement.

### CODE REVIEW

File: packages/desktop-ui/tsconfig.lib.json [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-    "types": []
+    "baseUrl": ".",
+    "outDir": "dist",
+    "declarationDir": "dist",
+    "paths": {
+      "@deepkit/core": ["../core/src/index.ts"],
+      "@deepkit/core-rxjs": ["../core-rxjs/src/index.ts"],
+      "@deepkit/event": ["../event/src/index.ts"],
+      "@deepkit/type": ["../type/src/index.ts"]
+    }
```
Comment (marcus-sa) [inline packages/desktop-ui/tsconfig.lib.json:12]: `ng-packagr` doesn't work with TypeScript 5.2.2 and project references

Comment (marcj) [reply packages/desktop-ui/tsconfig.lib.json:12]: we don't use ng-packagr anyways

Comment (marcus-sa) [reply packages/desktop-ui/tsconfig.lib.json:12]: @marcj we use ng-packagr for building the this package

Comment (marcj) [reply packages/desktop-ui/tsconfig.lib.json:12]: Oh I see, that was recently introduced. The library was designed to be TS only, so we could probably remove that again if it causes any issues

**Comment by marcus-sa** [comment]
wait for https://github.com/microsoft/TypeScript/issues/56480

**Pull request closed without merging**